lineman/app/components/dashboard_page/dashboard_page_controller.coffee

```diff
@@ -5,52 +5,57 @@ angular.module('loomioApp').controller 'DashboardPageController', ($rootScope, $
 
   Records.votes.fetchMyRecentVotes()
 
-  @perPage = 25
+  @perPage = 50
   @loaded =
     show_all:           0
     show_muted:         0
-    show_proposals:     0
     show_participating: 0
 
+  @views =
+    recent: {}
+    groups: {}
+
   @timeframes =
-    today:     { from: '1 second ago', to: '-10 year ago' }
+    today:     { from: '1 second ago', to: '-10 year ago' } # into the future!
     yesterday: { from: '1 day ago',    to: '1 second ago' }
     thisweek:  { from: '1 week ago',   to: '1 day ago' }
     thismonth: { from: '1 month ago',  to: '1 week ago' }
     older:     { from: '3 month ago',  to: '1 month ago' }
   @timeframeNames = _.map @timeframes, (timeframe, name) -> name
-  @timeframeQueryFor = (name) -> @timeframes[name].view
 
-  @groups = {}
+  @recentViewNames = ['proposals', 'starred', 'today', 'yesterday', 'thisweek', 'thismonth', 'older']
+
   @groupThreadLimit = 5
   @groups = -> CurrentUser.parentGroups()
-  @groupName = (group) -> group.name
-  @groupQueryFor = (group) -> @groups[group.key]
-  @moreForThisGroup = (group) -> @groupQueryFor(group, { filter: @filter }).length() > @groupThreadLimit
+  @moreForThisGroup = (group) -> @views.groups[group.key].length() > @groupThreadLimit
 
   @displayByGroup = ->
-    _.contains ['show_starred', 'show_muted'], @filter
+    _.contains ['show_muted'], @filter
 
-  @updateQueries = ->
-    _.each @groups(), (group) =>
-      @groups[group.key] = ThreadQueryService.groupQuery(group, { filter: @filter })
-    _.each @timeframeNames, (name) =>
-      @timeframes[name].view = ThreadQueryService.timeframeQuery
-        name: name
-        filter: @filter
-        timeframe: @timeframes[name]
+  @updateQueries = =>
+    if @displayByGroup()
+      _.each @groups(), (group) =>
+        @views.groups[group.key] = ThreadQueryService.groupQuery(group, { filter: @filter, queryType: 'all' })
+    else
+      @views.recent.proposals = ThreadQueryService.filterQuery ['show_proposals', @filter], queryType: 'important'
+      @views.recent.starred   = ThreadQueryService.filterQuery ['show_starred', 'hide_proposals', @filter], queryType: 'important'
+      _.each @timeframeNames, (name) =>
+        @views.recent[name] = ThreadQueryService.timeframeQuery
+          name: name
+          filter: @filter
+          timeframe: @timeframes[name]
 
   @loadMore = =>
     from = @loaded[@filter]
     @loaded[@filter] = @loaded[@filter] + @perPage
 
-    Records.discussions.fetchDashboard
+    Records.discussions.fetchDashboard(
       filter: @filter
       from:   from
-      per:    @perPage
+      per:    @perPage).then @updateQueries
   LoadingService.applyLoadingFunction @, 'loadMore'
 
-  @refresh = ->
+  @refresh = =>
     @updateQueries()
```

Comment:
Perhaps. I might not have my head entirely around when loki should and
shouldn't be updating.

An interesting example of that would be that when I changed it from the
previous implementation to purely views, it wouldn't live update the views
when they changed (ie, I click 'unstar' on a starred thread, and it doesn't
drop out of the starred box on dashboard until I revisit the page)

I manually fixed that one with a broadcast, but you're right that I can't
help but think there's something else that should be going on.


On Thu, Aug 13, 2015 at 12:45 PM, Robert Guthrie <notifications@github.com>
wrote:

> In lineman/app/components/dashboard_page/dashboard_page_controller.coffee
> <https://github.com/loomio/loomio/pull/2369#discussion_r37016955>:
>
> >    LoadingService.applyLoadingFunction @, 'loadMore'
> >
> > -  @refresh = ->
> > +  @refresh = =>
> >      @updateQueries()
>
> Hey James, just looking back over this. It's a really awesome piece of
> work.
>
> So I keep wondering when a bug is going to come up with how we update
> records.
> LokiJS has this update(obj) function docs <http://lokijs.org/#/docs>
> which it looks like you're supposed to call to tell it to rebuild indexes
> and views.
>
> So I wonder if you have to call updateQueries() here because we're failing
> to tell Loki that we've updated a record earlier on.
>
> Do you feel like that is the case, or is something else going on here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2369/files#r37016955>.
>